pkgs/development/python-modules/qutip/default.nix

```
@@ -0,0 +1,57 @@
+{ lib, fetchFromGitHub, python3Packages }:
+
+with python3Packages; buildPythonPackage rec {
+  pname = "qutip";
+  version = "4.6.2";
+
+  src = fetchFromGitHub {
+    owner = pname;
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "04g7ixq1yrrid4lliqbcamnzyw5r0fjbl8ipklps234hvsjfwmxb";
+  };
+
+  # QuTiP says it needs specific (old) Numpy versions. We overwrite them here
+  # as the tests work perfectly fine with up-to-date packages.
+  patches = [
+    ./requirements.patch
+  ];
+
+  # Disabling OpenMP support on Darwin.
+  setupPyGlobalFlags = lib.optional (!stdenv.isDarwin) "--with-openmp";
+
+  propagatedBuildInputs = [
+    packaging
+    numpy
+    cython
+    scipy
+    matplotlib
+  ];
+
+  doCheck = true;
+
+  checkInputs = [
+    pytest
+    pytest-rerunfailures
+  ];
+
+  # - QuTiP tries to access the home directory to create an rc file for us.
+  # This of course fails and therefore, we provide a writable temp dir as HOME.
+  # - We need to go to another directory to run the tests from there.
+  # This is due to the Cython-compiled modules not being in the correct location
+  # of the source tree.
+  # - For running tests, see:
+  # https://qutip.org/docs/latest/installation.html#verifying-the-installation
+  checkPhase = ''
```

The test phase scales up pretty big w.r.t. to the number of used CPUs, which can lead to trouble on build nodes with a lot of cores. Limiting the numbers of cores might be a good idea:
```suggestion
  checkPhase = ''
    export OMP_NUM_THREADS=$NIX_BUILD_CORES
```